Java/parkservices/src/main/java/com/amazon/randomcutforest/parkservices/calibration/ErrorHandler.java

```diff
@@ -0,0 +1,519 @@
+                    double offset = (2 - a) * lastDataDeviations[y % inputLength];
+                    errorRMSE.high[y] = (offset + (a - 1) * rmseHighDeviations[y].getDeviation());
+                    errorRMSE.low[y] = (offset + (a - 1) * rmseLowDeviations[y].getDeviation());
+                }
+            } else {
+                for (int y = 0; y < errorRMSE.high.length; y++) {
+                    errorRMSE.high[y] = rmseHighDeviations[y].getDeviation();
+                    errorRMSE.low[y] = rmseLowDeviations[y].getDeviation();
+                }
+            }
+
+            if (estimator != null) {
+                for (int i = 0; i < forecastHorizon; i++) {
+                    int[] missing = new int[inputLength];
+                    query[lastInputs.length] = i;
+                    query[lastInputs.length + 1] = forecastHorizon - i;
+                    for (int j = 0; j < inputLength; j++) {
+                        missing[j] = lastInputs.length + 2 + j;
+                    }
+                    // at this moment we use the PredictiveRCF more for the shorter term estimation,
+                    // and use an interpolation
+                    // with the observed error for the longer term
+                    SampleSummary answer = estimator.predict(query, 0, missing, 1, 0.5, 0.7);
+                    for (int j = 0; j < inputLength; j++) {
+
+                        errorHigh[i * inputLength + j] = (forecastHorizon - i)
+                                * max(0, answer.deviation[lastInputs.length + 2 + j]) / forecastHorizon
+                                + (float) (i * rmseHighDeviations[i * inputLength + j].getDeviation()
+                                        / forecastHorizon);
+
+                    }
+                    for (int j = 0; j < inputLength; j++) {
+                        missing[j] = lastInputs.length + inputLength + 2 + j;
+                    }
+                    answer = estimator.predict(query, 0, missing, 1, 0.5, 0.7);
+                    for (int j = 0; j < inputLength; j++) {
+
+                        errorLow[i * inputLength + j] = (forecastHorizon - i)
+                                * max(0, answer.deviation[lastInputs.length + inputLength + 2 + j]) / forecastHorizon
+                                + (float) (i * rmseLowDeviations[i * inputLength + j].getDeviation() / forecastHorizon);
+
+                    }
+                }
+            } else {
+                for (int y = 0; y < errorRMSE.high.length; y++) {
+                    errorHigh[y] = (float) errorRMSE.high[y];
+                    errorLow[y] = (float) errorRMSE.low[y];
+                }
+            }
+        }
+        // a control loop
+        for (int y = 0; y < intervalPrecision.length; y++) {
+            if (intervalPrecision[y].getMean() < 1.0 - percentile) {
+                errorHigh[y] = (float) max(1.0, 1.0 / (intervalPrecision[y].getMean() + 0.1)) * errorHigh[y];
```

Comment:
Should we add percentile instead of hardcoded 0.1 ?

Reply:
No No -- the hardcoded 0.1 is a "step size" -- if the interval precision was ruined (data shift or something) then it would not be great to make large changes when intervalPrecision[y] (approx) == 0.